### PR TITLE
Adding tests for [OZ] ERC-20 Votes contract

### PR DESCRIPTION
**Pull request by stefan-stefanooov**
**Description**:
Adding tests for ERC-20 Votes contract

**Related issue(s)**:
Add OZ erc20votes example #545

### CODE REVIEW

File: contracts/solidity/oz/ERC-20-votes/erc20VotesTest.sol [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: UNLICENSED
```
Comment (quiet-node) [inline contracts/solidity/oz/ERC-20-votes/erc20VotesTest.sol:1]: The license should be Apache-2.0

File: contracts/solidity/oz/ERC-20-votes/erc20VotesTest.sol [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+contract ERC20VotesTest is ERC20, ERC20Votes {
+    constructor(uint initialMintAmmount) ERC20("MyToken", "MYT") ERC20Permit("MyToken") {
+        _mint(msg.sender, initialMintAmmount);
+    }
+
+    function _afterTokenTransfer(address from, address to, uint256 amount)
+        internal
+        override(ERC20, ERC20Votes)
+    {
+        super._afterTokenTransfer(from, to, amount);
```
Comment (quiet-node) [inline contracts/solidity/oz/ERC-20-votes/erc20VotesTest.sol:17]: With the newest OZ v5.x, the `_beforeTokenTransfer` and `_afterTokenTransfer` hooks are deleted. I believe the new hook `_update` is supposed to be used instead. More info here https://github.com/OpenZeppelin/openzeppelin-contracts/releases/tag/v5.0.0

File: contracts/solidity/oz/ERC-20-votes/erc20VotesTest.sol [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    function _mint(address to, uint256 amount)
+        internal
+        override(ERC20, ERC20Votes)
+    {
+        super._mint(to, amount);
```
Comment (quiet-node) [inline contracts/solidity/oz/ERC-20-votes/erc20VotesTest.sol:24]: internal _mint is no longer virtual. Not sure if it makes a difference here but please check with the latest OZ version

File: contracts/solidity/oz/ERC-20-votes/erc20VotesTest.sol [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    function _burn(address account, uint256 amount)
+        internal
+        override(ERC20, ERC20Votes)
+    {
+        super._burn(account, amount);
```
Comment (quiet-node) [inline contracts/solidity/oz/ERC-20-votes/erc20VotesTest.sol:31]: internal _burn is no longer virtual

**Review by quiet-node — CHANGES_REQUESTED** [review]
The OZ@5.0.0 has a lot of significant changes. Please look at there [release note](https://github.com/OpenZeppelin/openzeppelin-contracts/releases/tag/v5.0.0). Also, my WIP branch is now all fixed. It is not ready to be merged but you should branch off from it so you can get the latest OZ version

File: contracts/solidity/oz/ERC-20-votes/erc20VotesTest.sol [hunk 5]
```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MIT
```
Comment (quiet-node) [inline contracts/solidity/oz/ERC-20-votes/erc20VotesTest.sol:1]: I believe we use Apache-2.0 for all Solidity licenses

File: test/solidity/oz/erc20votes/erc20Votes.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+        const signers = await ethers.getSigners()
+        wallet = signers[0]
+        wallet2 = signers[1]
+        const votesFactory = await ethers.getContractFactory(Contract.OZ.ERC20VotesTest)
+        contract = await votesFactory.deploy(AMOUNT_TO_MINT, {
+            gasLimit: 8000000,
+        })
+    })
+
+    it('should should deploy contract', async function () {
```
Comment (quiet-node) [inline test/solidity/oz/erc20votes/erc20Votes.js:43]: nit: the word "should" is duplicated

**Review by quiet-node — CHANGES_REQUESTED** [review]
LG just need to change the Solidity License and 1 nit.

File: test/solidity/oz/erc20votes/erc20Votes.js [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        const deployed = await contract.deployed()
+        expect(deployed).to.exist
+    })
+
+    it('should create/mint the erc20 tokens', async function () {
```
Comment (Nana-EC) [inline test/solidity/oz/erc20votes/erc20Votes.js:48]: Q: is this test doing the intended logic? Doesn't seem to match the description

File: test/solidity/oz/erc20votes/erc20Votes.js [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+            contract.getPastVotes(wallet.address, timeTick + TIME_INCREMENT)
+        ).to.eventually.be.rejected.and.have.property('errorName', FUTURE_LOOKUP_ERROR)
+    })
+
+    it('should produce an error when looking up tottle supply in the future (getPastTotalSupply) ', async function () {
+        const timeTick = await contract.clock()
+        const supply = await contract.getPastTotalSupply(timeTick - 1)
+        expect(supply).to.equal(AMOUNT_TO_MINT)
+    })
+
```
Comment (Nana-EC) [inline test/solidity/oz/erc20votes/erc20Votes.js:120]: Q: why this extra line before the closing bracket?

File: test/solidity/oz/erc20votes/erc20Votes.js (hunk 7 above)
Comment (stefan-stefanooov) [reply test/solidity/oz/erc20votes/erc20Votes.js:48]: The minting is done in the Smart Contracts constructor. 
I am just checking if we have the tokens.

File: test/solidity/oz/erc20votes/erc20Votes.js (hunk 8 above)
Comment (stefan-stefanooov) [reply test/solidity/oz/erc20votes/erc20Votes.js:120]: It is removed

File: test/solidity/oz/erc20votes/erc20Votes.js (hunk 7 above)
Comment (Nana-EC) [reply test/solidity/oz/erc20votes/erc20Votes.js:48]: The description should match the logic then

Comment (Nana-EC) [reply test/solidity/oz/erc20votes/erc20Votes.js:48]: But then you should check if it's a duplicate of the next test

**Review by Nana-EC — COMMENTED** [review]
LG
Please update comments and make sure tests are unique and isolated.

**Comment by stefan-stefanooov** [comment]
> LG just need to change the Solidity License and 1 nit.

@quiet-node ... Thank you for the review.
Can you please have a look ?

**Review by quiet-node — APPROVED** [review]
LGTM

**Pull request merged**